src/hotspot/share/runtime/mutexLocker.cpp

```diff
@@ -372,3 +372,22 @@ void print_owned_locks_on_error(outputStream* st) {
   }
   if (none) st->print_cr("None");
 }
+
+void unlock_locks_on_error(JavaThread* thread) {
```

Comment:
Does this have to live in mutexLocker.cpp? Seems to be highly specific to error handling and would make more sense as a function local static in vmError.cpp.

Reply:
`unlock_locks_on_error` refers to `_mutex_array`, which is a static variable.